great_expectations/expectations/metrics/column_aggregate_metrics/column_partition.py

```diff
@@ -155,8 +178,38 @@ def _get_column_partition_using_metrics(bins: int, n_bins: int, _metrics: dict)
                     int(np.ceil(sturges)), int(np.ceil(float(max_ - min_) / fd))
                 )
 
-        bins = np.linspace(start=float(min_), stop=float(max_), num=n_bins + 1).tolist()
+        bins = _determine_bins_using_proper_units(
+            datetime_detected=datetime_detected,
+            n_bins=n_bins,
+            min_=min_original,
+            max_=max_original,
+        )
     else:
         raise ValueError("Invalid parameter for bins argument")
 
     return bins
+
+
+def _determine_bins_using_proper_units(
+    datetime_detected: bool, n_bins: int, min_: Any, max_: Any
+) -> List[Any]:
+    if datetime_detected:
+        data: np.ndaarray = convert_ndarray_to_datetime_dtype_best_effort(
+            data=[min_, max_]
+        )
+        min_ = data[0]
+        max_ = data[1]
+        if n_bins == 0:
+            bins = [min_]
+        else:
+            delta_t = (max_ - min_) / n_bins
+            bins = []
+            for idx in range(n_bins + 1):
+                bins.append(min_ + idx * delta_t)
```

Comment:
And here we have the manual implementation of the calculation of datetime `bins` (ie. Evenly spaced bins, spaced apart by `delta_t`). It is able to go all the way down to seconds.